paper/JMNotes.Rmd

```diff
@@ -0,0 +1,45 @@
+# remarks JM
+
+- abstract: it is at least uncommon to cite other works in the abstract.
+If not explicitly required by the journal, I would refrain from it since you most likely cite all these works in the paper itself.
+- The last sentence of the abstract does not add much.
+Delete it or rewrite in a more compelling way.
+
+- I like the crop classification example, however, it is not entirely clear to me what agricultural districts are. 
+And though the crop classification introduces non-standard spatial resampling strategies, it does not make reference to the more common spatial resampling procedure (k-means clustering) where clustered coordinates are considered, which might be a more gentle introduction to spatial CV for readers who are not already familiar with the concepts.
+But I might be wrong.
+Looking at Table 1, the spatial blocking/tiles approach seems to be even more popular than the kmeans clustering approach.
+
+- Table 1: Wow, not too many papers have so far used spatial CV.
+Is this a comprehensive list?
+Have you somewhere listed the references by chosen strategy?
+If so, put it in the Appendix and reference the Appendix in the caption.
+Not entirely clear, what the footnotes refer to.
+These seem to be studies which have used the corresponding spatial CV approach but it is not all of them (e.g., spcv_block has been used by 28 papers but only 3 are referenced). 
+Please clarify.
+
+- I really do like the visualizations, great job! And they help the reader a lot to better grasp the differences between the resampling strategies.
+
+- In case, you didn't know: from 1.0.0 onwards, sf uses s2 by default, i.e. spherical geometries.
+Hence, I am not sure if there is really much more work to do (see https://r-spatial.github.io/sf/articles/sf7.html).
+I am referring to this sentence in the paper:
+
+>It is planned to provide a more consistent interface that supports all CRS and allows the use of distance measures in projected planar space or using great circle distances on the unprojected ellipsoid, in the case of unprojected CRS.
+
+- The leave-one-disc-out method seems to be quite promising since the direct neighbors between the training and test sets are removed thus removing spatial autocorrelation which might still be present when using the kmeans clustering approach.
+In the back of my mind, I knew that this was possible but I didn't know that this was already implemented in **sperrorest**.
+I wonder why AB has often preferred the kmeans clustering approach over the leave-one-disc-out approach.
+Maybe something which can be discussed.
+
+- Figure 7: spatial tiling. 
+It would be great if one could see the used rectangular blocks.
+- Figure 9: in fact, one only sees 4 time levels, not 5 as stated in the caption
+- LLTO: maybe you could expand a little bit on why observations need to be omitted.
+For readers new to the LLTO concept, the reasoning for this might not be entirely clear.
```

Comment:
I found Hanna's [conceptual figure](https://github.com/mlr-org/mlr3spatiotempcv/files/4142419/cv.pdf) quite enlightening, and I think it also does a better job of showing why and which observations have to be omitted than Figure 11 (it's just too many observations). Here's my take on the explanation (not sure if its easier to understand):
`In order to make a prediction for a specific time stamp of a specific location (test set), one would have to omit both the observations of the same location as well a the observations from the same time stamp of other locations in the training set.`

Reply:
> In order to make a prediction for a specific time stamp of a specific location (test set), one would have to omit both the observations of the same location as well a the observations from the same time stamp of other locations in the training set.

Why "have to omit"? There's nothing you have to remove by default. If my prediction task is to predict a future outcome at a location based on that location's history, then it doesn't make sense to remove its history from the training set. So let's refrain from telling the reader what they should be doing - we only offer technological solutions that may or may not fit their prediction tasks. We could use phrases such as "this resampling design may be particularly useful in situations in which..." to convince the reader that what was implemented may actually be useful in some relevant situation...

Reply:
Basically, I only wanted to describe what LLTO is doing but I see your point. Ok, then let's rephrase in a more neutral tone:
If there are multiple observations over time per location, and the modeling purpose is to reduce the bias in the predictions as potentially introduced by spatial as well as temporal autocorrelation, LLTO might be an appropriate choice. To do so, LLTO makes predictions for a specific time stamp of a specific location (test set) by omitting the observations of this specific location as well a the observations of this specific time stamp of other locations in the training set.

Reply:
> I found Hanna's conceptual figure quite enlightening, and I think it also does a better job of showing why and which observations have to be omitted than Figure 11 (it's just too many observations).

For me the figure never easily clicked in my head but I think it can be quite helpful for others!

> So let's refrain from telling the reader what they should be doing - we only offer technological solutions that may or may not fit their prediction tasks. We could use phrases such as "this resampling design may be particularly useful in situations in which..." 

Yes, wording is really the key issue in this paper - similar to the other comment above where we discussed about how to "get it right for the reader" to avoid making false promises or strong statements.

---

Here is the new paragraph

> The "Leave-Location-and-Time-Out" (LLTO) method combines both "Leave-Time-Out" (LTO) and "Leave-Location-Out" (LLO).
It creates partitions that operate on spatiotemporal clusters to reduce both spatial and temporal autocorrelation.
If there are multiple observations over time per location, and the modeling purpose is to reduce the bias in the predictions as potentially introduced by spatial as well as temporal autocorrelation, LLTO might be an appropriate choice.
To reduce autocorrelation in both dimensions, the methods omits observations as the splitting in test and train sets only provides separation in one dimension whereas two (space and time) are needed to account for both types, spatial and temporal autocorrelation.
In other words, accounting for one dimension as in Fig. \@ref(fig:lto) or Fig. \@ref(fig:llo), by only having one time step or location in either test or training set, would have no effect on the respective other.
Thus, the removal of observation can be seen as a proxied buffering in multiple dimensions as shown in Fig. \@ref(fig:llto).

Reply:
IMHO still hard to follow but that might be just personal taste, so leave it as is, and wait for potential reviewer comments.